server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -244,6 +244,7 @@ public void onReplicationDone(SegmentReplicationState state) {
                                 runnable.run();
                             }
                         }
+                        replicaShard.refresh("replication complete refresh");
```

Comment:
@Rishikesh1159 Why is this extra refresh required? The reader should refresh before replication is marked completed.